File: annuaire_entreprise_front/src/utils/passwordValidator.js
export const validatePassword = (password) => {
    let errors = "";
  
    // Vérification de la longueur (min 7 caractères)
    if (password.length < 7) {
        errors = "Le mot de passe doit contenir au moins 7 caractères.";
    }
    // Vérification d'une lettre majuscule
    else if (!/[A-Z]/.test(password)) {
        errors = "Le mot de passe doit contenir au moins une lettre majuscule.";
    }
    // Vérification d'un caractère spécial
    else if (!/[^a-zA-Z0-9]/.test(password)) {
        errors = "Le mot de passe doit contenir au moins un caractère spécial (!@#$%^&* etc.).";
    }
  
    // Si aucune erreur, retourner l'état valide
    return errors.length === 0 ? { valid: true } : { valid: false, errors };
  };
  

File: API_annuaire_entreprise/Shared/Validators/PasswordValidator.cs
using System.ComponentModel.DataAnnotations;
using System.Text.RegularExpressions;

namespace API_annuaire.Shared.Validators;

public class PasswordValidator : ValidationAttribute
{
    public override bool IsValid(object? value)
    {
        var input = value?.ToString();

        var messages = new List<string>();

        ErrorMessage = string.Empty;

        if (string.IsNullOrEmpty(input))
        {
            ErrorMessage = "Le mot de passe doit être renseigné";
            return false;
        }

        var hasUpperLetter = new Regex(@"[A-Z]");
        var hasEnoughChars = new Regex(@".{7,}");
        var hasSymbol = new Regex(@"[^a-zA-Z0-9]");

        if (!hasUpperLetter.IsMatch(input))
        {
            messages.Add("Il manque une lettre majuscule.");
        }

        if (!hasEnoughChars.IsMatch(input))
        {
            messages.Add("Le mot de passe doit contenir au moins 7 caractères.");
        }

        if (!hasSymbol.IsMatch(input))
        {
            messages.Add("Il manque un caractère spécial.");
        }

        ErrorMessage = string.Join("\n", messages);

        return string.IsNullOrEmpty(ErrorMessage);
    }
}
